### ISSUE TITLE
AZ-800 Lab 6, Exercise 1

### ISSUE THREAD

**Issue by mikegrant5** [MicrosoftLearning/AZ-800-Administering-Windows-Server-Hybrid-Core-Infrastructure#68]
# Module/Lab: 06
#### Step: 02

Description of issue
Cannot activate Defender for Cloud by clicking "Upgrade"
Repro steps:

2.	From the Microsoft Defender for Cloud | Getting started page, enable enhanced security of Microsoft Defender for Cloud and enable automatic Microsoft Defender for Cloud agent installation
1.
Reason:  Microsoft seem to have changed all trials (this would be a 30 day trial) to demand billing details before allowing the trial - this means in a Cloud Slice lab, that no student (or instructor) is going to be able to do that, or want to do that.

This also spoils later parts of the lab

**Comment by dvimacedo** [comment]
Hi @mikegrant5, thank you for the feedback on the lab.
Could you give some more information regarding the environment where you are running the lab?

**Comment by mikegrant5** [comment]
I have found what the problem is:


(i)                  It seems to be a problem when generating the template here (you do not get the problem, if you pre-save the template files just before deploying the original vm)


***@***.***

When using the template.json and parameters.json files (generated by the above) in a "custom template" deployment, if you select "Edit parameters" having loaded and saved the template.json, then you get the below, but if you then choose "load file" and select the parameters.json file, then it loads the blank one (shown above), whereas the one above, used to contain pre-filled values, which you could then change on subsequent deployments.

***@***.***

(ii)                Deploying with the above parameters (I changed the names as the existing VM would clash, and it still fails as per below:

***@***.***
***@***.******@***.***

(iii)               Same thing happens if I load the parameters.json (which then inherits the values from the template.json anyway)

(iv)               If I try a custom template deployment with the previously saved .json files from when I created the original vm (just prior to deploying it), it all works just fine.  This seems to be because the original json files have a complete specification, whereas the ones created by exporting from the resource group are much shorter and do not contain what is needed for a successful deployment.

So, in summary, it seems that the Azure Portal menus have introduced a fault in not generating the correct template.
As stated above, you do not get the problem, if you pre-save the template files just before deploying the original vm.

I have attached the two sets of template and parameters json files; the ones with the RG_ prefix come from exporting the template files from the resource group.

Hope this helps


Mike

**Comment by dvimacedo** [comment]
Skillable environment has the trials already activated, added note inside environment to clear further issues.
